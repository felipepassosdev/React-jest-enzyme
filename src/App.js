import React, { Component } from "react";
import Account from "./Account";

class App extends Component {
  state = {
    isLoading: true,
    users: [],
    error: null,
  };

  fetchUsers() {
    fetch(`https://jsonplaceholder.typicode.com/users`)
      .then((response) => response.json())
      .then((data) => {
        this.setState({
          users: data,
          isLoading: false,
        })
      })
      .catch((error) => this.setState({ error, isLoading: false }));
  }

  componentDidMount() {
    this.fetchUsers();
  }

  render() {
    const { isLoading, users, error } = this.state;
    return (
      <>
        <h1>title</h1>
        <p>First paragraph</p>
        <h2>Welcome to React Testing</h2>;
        <button type="button" id="my-button-two"></button>
        <button type="button" id="my-button-three"></button>
        {error ? <p>{error.message}</p> : null}
        {!isLoading ? (
          users.map((user) => {
            return <Account key={user.username} user={user}/>;
          })
        ) : (
          <h3>Fetching Users...</h3>
        )}
      </>
    );
  }
}

export default App;
