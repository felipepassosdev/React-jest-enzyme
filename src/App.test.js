import React from "react";
import { shallow, mount } from "enzyme";
import Account from "./Account";
import App from "./App";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const wrapper = mount(<App />);
  expect(wrapper.state("error")).toEqual(null);
});


 it("renders without crashing", () => {
  shallow(<App />);
});

it("renders Account header", () => {
  const wrapper = shallow(<App />);
  const header = <h1>title</h1>;
  expect(wrapper.contains(header)).toEqual(true);
}); 


it("renders paragraph", () => {
  const wrapper = shallow(<App />);
  const header = <p>First paragraph</p>;
  expect(wrapper.contains(header)).toEqual(true);
}); 

it("renders h3", () => {
  const app = shallow(<App/>);
  const title = <h3>Fetching Users...</h3>;
  expect(app.contains(title)).toEqual(true)
});

it ("renders h3", () => {
  const wrapper = shallow(<App/>);
  const value = wrapper.find("h3").text();
  expect(value).toEqual("Fetching Users...")
});

it ("render button", () => {
  const wrapper = shallow(<App/>);
  wrapper.find('button#my-button-three').simulate("click")
})

const user = {
  name: "Adeneye David",
  email: "david@gmail.com",
  username: "Dave"
};

describe("<Account />", () => {
  it("contains account", () => {
    const wrapper = mount(<Account user={user} />);
    const value = wrapper.find("p").text();
    expect(value).toEqual("david@gmail.com");
  });

  it("accepts user account props", () => {
    const wrapper = mount(<Account user={user} />);
    expect(wrapper.props().user).toEqual(user);
  });

  it("accepts phone account props", () => {
    const wrapper = mount(<Account user={user}/>);
    expect(wrapper.props())
  })
});

 it("renders welcome message", () => {
  const wrapper = shallow(<App />);
  const welcome = <h2>Welcome to React Testing</h2>;
  expect(wrapper.contains(welcome)).toBe(true);
  expect(wrapper.contains(welcome)).toEqual(true);
});


 
 

