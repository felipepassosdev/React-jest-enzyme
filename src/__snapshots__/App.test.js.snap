// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <h1>
    Display Active Users Account Details
  </h1>
  <h3>
    Fetching Users...
  </h3>
</Fragment>
`;

exports[`renders without crashing 1`] = `
<Fragment>
  <h1>
    title
  </h1>
  <p>
    First paragraph
  </p>
  <h2>
    Welcome to React Testing
  </h2>
  ;
  <h3>
    Fetching Users...
  </h3>
</Fragment>
`;
